superdesk/text_checkers/ai/semaphore.py

```diff
@@ -0,0 +1,472 @@
+import os
+import logging
+import requests
+import xml.etree.ElementTree as ET
+from flask import current_app, abort
+from .base import AIServiceBase
+import traceback
+import io
+import json
+
+
+
+logger = logging.getLogger(__name__)
+session = requests.Session()
+
+TIMEOUT = (5, 30)
+
+
+class Semaphore(AIServiceBase):
+    """Semaphore autotagging service
+    
+    Environment variables SEMAPHORE_BASE_URL, SEMAPHORE_ANALYZE_URL, SEMAPHORE_SEARCH_URL, SEMAPHORE_GET_PARENT_URL and SEMAPHORE_API_KEY must be set.
+    """
+
+    name = "semaphore"
+    label = "Semaphore autotagging service"
+    print(AIServiceBase)
+
+
+	
+	
+    def __init__(self,data):
+
+	# SEMAPHORE_BASE_URL OR TOKEN_ENDPOINT Goes Here
+        self.base_url =  os.getenv('SEMAPHORE_BASE_URL')
+
+	#  SEMAPHORE_ANALYZE_URL Goes Here
+        self.analyze_url = os.getenv(' SEMAPHORE_ANALYZE_URL')
+
+	#  SEMAPHORE_API_KEY Goes Here
+        self.api_key = os.getenv('SEMAPHORE_API_KEY')
+
+	#  SEMAPHORE_SEARCH_URL Goes Here
+        self.search_url = os.getenv('SEMAPHORE_SEARCH_URL')
+
+	#  SEMAPHORE_GET_PARENT_URL Goes Here
+        self.get_parent_url = os.getenv('SEMAPHORE_GET_PARENT_URL')
+
+        
+
+        self.session = requests.Session()  
```

Comment:
seems like this is not used later, so you can avoid that or use it later instead of the session defined outside of the class